source/elements/oneDAL/source/glossary.rst

```diff
@@ -0,0 +1,170 @@
+
+    Observation
+        A vector of at least one :term:`feature` value and zero or more :term:`label<Label>` values.
+        Each of :term:`labels<Label>` and :term:`features<Feature>` can encode different objects or events.
+
+        Synonyms: instance, sample.
+
+    Ordinal feature
+        A :term:`feature` with a discrete set of possible values and some ordering between them.
+
+    Regression
+        A :term:`supervised machine-learning problem<Supervised learning>` of assigning
+        :term:`continuous<Continuous feature>` :term:`labels<Label>` to
+        :term:`feature vectors<Feature vector>`.
+
+    Response
+        A value of a single :term:`label`. For example, response may be a flag
+        indicating that an observation belongs to a particular class.
+
+    Supervised learning
+        :term:`Training` process with :term:`labeled<Label>` :term:`training set`.
+        The best :term:`model` of dependencies between input :term:`feature vector`
+        and :term:`labels<Label>` is selected.
+
+    Training
+        A process of creating a best :term:`model` (in terms of some quality criteria) based on
+        information extracted from a :term:`training set`.
+
+    Training set
+        A :term:`dataset` used at the :term:`training` stage to create a :term:`model`.
+
+    Unsupervised learning
+        :term:`Training` process where :term:`training set` does not contain information
+        of dependencies between :term:`feature vectors<Feature vector>` and outputs need
+        to be produced by the :term:`model`.
+
+|dal_short_name| terms
+======================
+
+.. glossary::
+    :sorted:
+
+    Batch Mode
+        The computation mode for an algorithm in |dal_short_name|, where all the data needed for
+        computation is available at the start and fits the memory of the device on which the computations are performed.
+
+    Contiguous data
+        A special property of the :term:`data format`: the data are stored as one contiguous memory block.
+
+    Data format
+        The representation of internal structure of the data: this information is enough to efficiently access them.
+
+    Data type
+        An integral or floating-point type which is used to physically store the data.
+
+    Homogeneous data
+        A special property of the :term:`data format`: all the :term:`features<Feature>` have the same type and domain.
+
+    Immutability
+        The object is immutable if it is not possible to change its state after creation.
+
+    Online Mode
+        The computation mode for an algorithm in |dal_short_name|, where the data needed for computation
+        becomes available in parts over time.
+
+    Reference-counted object
+        A |dal_short_name| object which stores the number of references to the unique implementation.
+        Such object has a lightweight copy constructor and assignment operator, which simply increases the number of references.
```

Comment:
it requires def of copy constructor/assignment operator - is it possible to simplify?
'simply' - remove

Reply:
Reformulated with C++ terminology - copy-assignable and copy-constructible. Thanks!